### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by MaksymMaliuk**
[DEMO LINK](https://MaksymMaliuk.github.io/react_clock/)

### CODE REVIEW

**Review by kateryna-babina — APPROVED** [review]
GJ

File: src/App.tsx [hunk 1]
```diff
@@ -1,39 +1,62 @@
...
+    this.setState({ hasClock: false });
+  };
+
+  render() {
+    const { hasClock, clockName } = this.state;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        {hasClock && (
```
Comment (sergii-nosachenko) [inline src/App.tsx:56]:
```suggestion

        {hasClock && (
```

File: src/components/Clock.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  };
+
+  clockId = 0;
+
+  componentDidMount() {
+    this.clockId = window.setInterval(() => {
+      this.setState({ today: new Date() });
+
+      window.console.info(
+        formattedTime(this.state.today),
```
Comment (sergii-nosachenko) [inline src/components/Clock.tsx:27]: It will print a previous value, not a current

File: src/components/Clock.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+export class Clock extends React.Component<Props, State> {
+  state = {
+    today: new Date(),
+  };
+
+  clockId = 0;
+
+  componentDidMount() {
+    this.clockId = window.setInterval(() => {
```
Comment (sergii-nosachenko) [inline src/components/Clock.tsx:23]:
```suggestion
    this.clockId = setInterval(() => {
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,39 +1,62 @@
...
+export class App extends React.Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
```
Comment (sergii-nosachenko) [inline src/App.tsx:25]:
```suggestion
    this.timerId = setInterval(() => {
```

File: src/App.tsx [hunk 5]
```diff
@@ -1,39 +1,62 @@
...
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends React.Component<{}, State> {
+  state = {
```
Comment (sergii-nosachenko) [inline src/App.tsx:17]: Add Readonly type

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 6]
```diff
@@ -1,39 +1,63 @@
...
+    document.removeEventListener('click', this.handleLeftClick);
+    document.removeEventListener('contextmenu', this.handleRightClick);
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  handleLeftClick = (event: Event) => {
+    event.preventDefault();
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:41]: I guess there is no sense to prevent `click` event default behaviour

File: src/components/Clock.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+type State = {
+  today: Date;
+};
+
+type Props = {
+  clockName: string;
+};
+
+const formattedTime = (date: Date) => (
```
Comment (OleksiiBlezniuk) [inline src/components/Clock.tsx:11]:
```suggestion
const formatTime = (date: Date) => (
```

Comment (OleksiiBlezniuk) [reply src/components/Clock.tsx:11]: formatTime - function
formattedTime - result of this function

**Review by OleksiiBlezniuk — APPROVED** [review]

**Pull request closed without merging**